[CISupport][build.webkit.org][GTK] Add new packaging bot for GNOME Web Canary nightlies

clopez opened the pull request:
#### ccb2d5348d7e97c46c4b5f82d4db0ce34b2f143e
<pre>
[CISupport][build.webkit.org][GTK] Add new packaging bot for GNOME Web Canary nightlies
<a href="https://bugs.webkit.org/show_bug.cgi?id=282842">https://bugs.webkit.org/show_bug.cgi?id=282842</a>

Reviewed by Philippe Normand.

This adds a new packaging bot that will run once per day and will build WebKitGTK
with the GNOME SDK and then will upload the builtproduct via SFTP like the other
packaging bots do.

On top of that, this also removes the step install-built-product from all the bots
except this one. This step was added in 239394@main (556db83) and is really only
needed on this bot that is going to upload the built product.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories.py:
(BuildFactory.__init__):
(BuildAndTestAndArchiveAllButJSCFactory.__init__):
(BuildAndUploadBuiltProductviaSftpFactory):
(BuildAndUploadBuiltProductviaSftpFactory.__init__):
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit.start):
(ArchiveMinifiedBuiltProduct):
(UploadBuiltProductViaSftp):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_success_gtk):
* Tools/CISupport/ews-build/factories.py:
(BuildFactory.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(CompileWebKit.start):
(AnalyzeJSCTestsResults.send_email_for_pre_existing_failure):
(InstallBuiltProduct): Deleted.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/286354@main">https://commits.webkit.org/286354@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c0ffe11ab6b1c3c352a894dea2e2bc5b3613aed7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75741 "Passed style check") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28592 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80230 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27005 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/77857 "Passed tests") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3029 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/59405 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/17575 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78808 "Passed tests") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/65060 "Build was cancelled. Recent messages:OS: Ventura (13.6.7), Xcode: 14.3; Skipping applying patch since patch_id isn't provided; Checked out pull request; run-api-tests (cancelled)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39761 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22543 "Build is in progress. Recent messages:OS: Ventura (13.6.7), Xcode: 14.3") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/25334 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22880 "Build is in progress. Recent messages:OS: Ventura (13.6.7), Xcode: 14.3") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/81701 "Build was cancelled. Recent messages:Printed configuration") | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3080 "Build is in progress. Recent messages:OS: Sequoia (15.0.1), Xcode: 16.0") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/1955 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/81701 "Build was cancelled. Recent messages:Printed configuration") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/75418 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3238 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65027 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/81701 "Build was cancelled. Recent messages:Printed configuration") | 
| | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/10887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/9022 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11694 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3045 "Built successfully") | | | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3063 "Hash c0ffe11a for PR 36405 does not build (failure)") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/4003 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/3071 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Tools/CISupport/build-webkit-org/config.json:
```diff
@@ -541,6 +542,12 @@
+                  {
+                    "name": "GTK-Linux-64bit-Release-GNOMEWebCanary-Packaging-Nightly", "factory": "BuildAndUploadBuiltProductviaSftpFactory",
+                    "platform": "gtk", "configuration": "release", "architectures": ["x86_64"],
+                    "additionalArguments": ["--no-bubblewrap-sandbox", "--prefix=/app/webkit/WebKitBuild/release/install"],
```
```suggestion
                    "additionalArguments": ["--no-bubblewrap-sandbox", "--prefix=/app/webkit/WebKitBuild/Release/install"],
```

clopez replied:
Changed, I also removed `--no-bubblewrap-sandbox` so that we can catch easily any regression on the sandbox causing issues inside flatpak. I know it gets disabled automatically at runtime inside flatpak, so in order to test that it works.

philn reviewed (approved):
Thanks :)

clopez committed:
[CISupport][build.webkit.org][GTK] Add new packaging bot for GNOME Web Canary nightlies
https://bugs.webkit.org/show_bug.cgi?id=282842

Reviewed by Philippe Normand.

This adds a new packaging bot that will run once per day and will build WebKitGTK
with the GNOME SDK and then will upload the builtproduct via SFTP like the other
packaging bots do.

On top of that, this also removes the step install-built-product from all the bots
except this one. This step was added in 239394@main (556db83) and is really only
needed on this bot that is going to upload the built product.

* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories.py:
(BuildFactory.__init__):
(BuildAndTestAndArchiveAllButJSCFactory.__init__):
(BuildAndUploadBuiltProductviaSftpFactory):
(BuildAndUploadBuiltProductviaSftpFactory.__init__):
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit.start):
(ArchiveMinifiedBuiltProduct):
(UploadBuiltProductViaSftp):
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_success_gtk):
* Tools/CISupport/ews-build/factories.py:
(BuildFactory.__init__):
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/CISupport/ews-build/steps.py:
(CompileWebKit.start):
(AnalyzeJSCTestsResults.send_email_for_pre_existing_failure):
(InstallBuiltProduct): Deleted.
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: https://commits.webkit.org/286354@main
